### PR TITLE
[READY]pathfinding circuitry upgrade

### PR DESCRIPTION
**Pull request by arsserpentarium**
[Changelogs]: 
New circuitry update. Now your drones will be able to solve some cargo related tasks. like crate moving and material distribution.

:cl: selea
add: a* pathfinder, coordinate pathfinder(accepts not ref to some obj, but it's abs coordinates) ,turf pointer(can give you ref to turf with given coordinates), turf scanner(can read letters on floor and contents of turf),material scaner(can read amounts of materials in machinery),material manipulator(to load/unload resources)
tweak: now you can decide, if basic pathfinder should avoid obstacles, or not.Now demux push only desired output without unnecessary nulls,upgraded claw and local locator.
balance: reduced complexity cost of locomotors,abs/rel converters and basic pathfinders.Increased complexity of throwers, 
fix: Fixed bugs with cryptography, obstacle ref in locomotors, release pin in pullng claw.
refactor: add material tranfer proc to mat containers.refactored pathfinder SS to have separated queue for mobs and circuits to avoid spam.
/:cl:

### CODE REVIEW

**Comment by GunHog** [comment]
Is the circuit's ASTAR path displayed on the Diagnostic HUD like the other bots?

**Comment by arsserpentarium** [comment]
Most likely not

**Comment by Buggy123** [comment]
>Increased complexity of throwers,

A T O M I C

**Comment by juanototo2** [comment]
>Increased complexity of throwers,
THIS BOI BE GOIN **NUCLEAAAAAAAAAAR**

**Comment by arsserpentarium** [comment]
Why the fuck you concerned about throwers, when you have A*?

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 1]
```diff
@@ -457,3 +471,123 @@
 
 	A.forceMove(drop_location())
 	A.throw_at(locate(x_abs, y_abs, T.z), range, 3)
+
+/obj/item/integrated_circuit/manipulation/matman
```
Comment (PKPenguin321) [inline code/modules/integrated_electronics/subtypes/manipulation.dm]: nananananananana nananananananana,

**Comment by PKPenguin321** [comment]
biiiig 👍 hope it's ready to merge and merged soon
also https://github.com/tgstation/tgstation/issues/36646

**Comment by PKPenguin321** [comment]
I honestly can't wait to use the coord pathfinder aaaaaaaa

File: code/modules/integrated_electronics/subtypes/manipulation.dm (hunk 1 above)
Comment (arsserpentarium) [reply code/modules/integrated_electronics/subtypes/manipulation.dm]: Bananananana?

**Pull request merged**